### PR TITLE
Display a warning and gas fee component for token allowance and NFT flow when transaction is expected to fail

### LINKED ISSUE #17377: [Bug]: Gas fee is not displayed in some confirmation screens when transaction is expected to fail
### Describe the bug

When MM loads a confirmation screen for a transaction that is expected to fail, it's hiding the gas component and no gas information can be seen at all.

This has been seen so far on both the NFT approve and the new token allowance confirmation screens. There might be other confirmations where this might be happening as well. 

### Steps to reproduce

1. Go to (erc721) https://etherscan.io/address/0x2a187453064356c898cae034eaed119e1663acb8#writeContract or (erc1155) https://etherscan.io/address/0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d#writeContract
2. Connect your wallet
3. Go to approve and input a random address and a random token id (I used `1`, it's important that you don't actually own that NFT)
4. Click on `Write`
5. Notice that the MM popup does not show the gas component

@seaona might also have repro steps for the token allowance

### Error messages or log output

_No response_

### Version

10.24.1

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

Screenshot (MM v10.24.1):
Screenshot 2023-01-24 at 12 04 27

Screenshot (MM v10.23.1):
Screenshot 2023-01-24 at 12 18 58

### LINKED ISSUE #17379: [Bug]: Failing Transactions don't display a Warning anymore and set an insane amount of Gas fees
### Describe the bug

**Problem**: when a tx is going to fail, it displays an insane amount of Gas fees and no warning:

image

We used to display a warning, like this:

image

### Steps to reproduce

1. Select Mainnet
2. Go to https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7#writeContract
3. Connect MM
4. Click Deprecate
5. Add an address
6. Check MM

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

With this fixes, when MM loads a confirmation screen for a transaction that is expected to fail, it's now displaying the gas component. This has been seen so far on both the NFT approve and the new token allowance confirmation screens. Also when a transaction is expected to fail, it displays a warning. Approving a second time USDT drives to a failed transaction where should display a warning and the gas component. 

Further explanation:

The gas component isn't displayed because we aren't set `userAcknowledgedGasMissing` in `GasDetailsItem` and `EditGasFeeButton` component in `approve-content-card.js` (this is content for token allowance flow) and `confirm-approve-content.component.js` (this is for NFT flow). Default value for `userAcknowledgedGasMissing` is `false`.

Code where default value for `userAcknowledgedGasMissing` is `false` in `GasDisplayItem` component: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/gas-details-item/gas-details-item.js#L19
Code where it is returning `null` in `GasDisplayItem` component: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/gas-details-item/gas-details-item.js#L35

Code where it is returning `null` in `EditGasFeeButton` component: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/edit-gas-fee-button/edit-gas-fee-button.js#L35

Now we are using `renderSimulationFailureWarning` to show a warning if it is `true` and if we want to proceed with a transaction that is expected to fail, we are setting `userAcknowledgedGasMissing` to `true` when we click on `I want to proceed anyway` link and then gas component will displays.

I also changed `setshowContractDetails` to `setShowContractDetails` what @darkwing suggested [here](https://github.com/MetaMask/metamask-extension/pull/16740#discussion_r1084211180).

* Fixes #17379
* Fixes: #17377

#### ERC20 token (token allowance flow)

https://user-images.githubusercontent.com/92527393/214882232-ace59826-adc5-4945-9f11-bb0554c14494.mov

#### ERC721 or ERC1155 token (NFT flow)

https://user-images.githubusercontent.com/92527393/214882696-b1ee3add-c978-444e-82b2-8b3e4177efc2.mov

#### ERC20 token (token allowance flow)

https://user-images.githubusercontent.com/92527393/214870936-eb1cdb88-5f06-42f8-823e-80fb00fad167.mov

#### ERC721 or ERC1155 token (NFT flow)

https://user-images.githubusercontent.com/92527393/214871846-00285c51-9cdc-41be-98a1-d1bf1185e0aa.mov

### ERC20 token (token allowance)

1. Go to USDT https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7#writeContract
2. Click `Write Contract`
3. Connect MM
4. Click `Approve`
5. Set amount and address
6. Click on `Write`
7. Approve on MM
8. Click again Approve -- this transaction will fail. See how it is displays a warning and when user click on `I want to proceed anyway` link displays the gas component for this failing transaction (on the second page of token allowance flow)

### ERC721 or ERC1155 token (NFT flow)

1. Go to (erc721) https://etherscan.io/address/0x2a187453064356c898cae034eaed119e1663acb8#writeContract or (erc1155) https://etherscan.io/address/0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d#writeContract
2. Connect your wallet
3. Go to `approve` and input a random address and a random token id (I used 1, it's important that you don't actually own that NFT)
4. Click on `Write`
5. See how it is displays a warning and when user click on `I want to proceed anyway` link displays the gas component for this failing transaction

### CODE REVIEW

**Comment by danjm** [comment]
I think that another fix is required for the confirmation screens other than token allowance and NFT approvals. I have made a PR for that here: https://github.com/MetaMask/metamask-extension/pull/17458

**Comment by VSaric** [comment]
> 

Oh I see, sorry for that @danjm! Do you want maybe to delete that line in `confirm-transaction-base.component.js` in this PR or it would be in separate PR that you opened?

**Comment by dzfjz** [comment]
Verified by QA

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js [hunk 1]
```diff
@@ -85,15 +87,43 @@ export default class ConfirmApproveContent extends Component {
...
+                  ? undefined
+                  : {
+                      label: t('proceedWithTransaction'),
+                      onClick: () =>
+                        this.setState({ userAcknowledgedGasMissing: true }),
+                    }
+              }
+            />
+          </Box>
+        )}
```
Comment (jpuri) [inline ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js:126]: Might be useful to extract out a small component for above warning display. It seems to be repeated a few places.

**Comment by jpuri** [comment]
Will be nice to have more test coverage in the PR.

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js [hunk 2]
```diff
@@ -141,8 +176,16 @@ export default class ConfirmApproveContent extends Component {
...
+    const { userAcknowledgedGasMissing } = this.state;
+    const hasSimulationError = Boolean(txData.simulationFails);
+    const renderSimulationFailureWarning =
+      hasSimulationError && !userAcknowledgedGasMissing;
+    if (
+      !isMultiLayerFeeNetwork &&
+      supportsEIP1559 &&
+      !renderSimulationFailureWarning
+    ) {
+      return <GasDetailsItem userAcknowledgedGasMissing />;
```
Comment (jpuri) [inline ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js:188]: A re-usable hook to get this value `renderSimulationFailureWarning` might be useful. I see this logic repeated a few places.

Comment (VSaric) [reply ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js]: @jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17437/commits/cabe23bdddf6470922c9215baf57b396a4a8e953).

**Review by danjm — APPROVED** [review]

**Review by ryanml — APPROVED** [review]
lgtm!

File: ui/components/ui/simulation-error-message/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from './simulation-error-message';
```
Comment (jpuri) [inline ui/components/ui/simulation-error-message/index.js:1]: A better place for this will be `ui/components/app`

**Pull request merged**